### DISCUSSION TITLE
更換u50的platform

### DISCUSSION THREAD

**Discussion by ray3210ray3210** [bol-edu/HLS-SOC-Discussions#51] [General]
我們在Final project實作別人github的設計時，因為必須使用vitis 2021.1的版本。
遇到了以下的問題

02.Yhjyyx@HLS02:~/workspace_final$ vitis -version

****** Xilinx Vitis Development Environment
****** Vitis v2021.1 (64-bit)
  **** SW Build 3245906 on 2021-06-09-18:10:55
    ** Copyright 1986-2021 Xilinx, Inc. All Rights Reserved.

02.Yhjyyx@HLS02:~/workspace_final$ platforminfo -p xilinx_u50_gen3x16_xdma_5_202210_1
**Error** parsing platform file: /opt/xilinx/platforms/xilinx_u50_gen3x16_xdma_5_202210_1/xilinx_u50_gen3x16_xdma_5_202210_1.xpfm
Line number:16. Mandatory element missing
Exception: failed to parse the XPFM file

上網查詢之後發現vitis2021.1無法支援這個u50的platform,[here](https://docs.xilinx.com/r/en-US/ug1120-alveo-platforms/Alveo-Platforms)
**想請問我們有辦法更換u50的platform嗎?**

必須使用vitis2021.1的版本的原因是
他們的設計在2021.2之後的版本
做pipeline的時候會出現這個問題,[here](https://github.com/spcl/gemm_hls/issues/29)
我們已經在2022.1compile過，也會出現同樣的問題

Thank you.
許睿哲

**Comment by boledulab** [comment]
You can run two source commands to change to vitis 2021.1 on HLS02 machine. (default is vitis 2022.1)
source /opt/Xilinx/Vitis/2021.1/settings64.sh
source /opt/xilinx/xrt/setup.sh

**Reply by ray3210ray3210** [reply]
I followed the steps to change to the **Vitis 2021.1** version, but I'm still encountering the same problem,
![as shown](https://github.com/bol-edu/HLS-Discussions/assets/98219588/547b702a-74b2-49dc-b0f2-4c2da5a48f57)

I found someone who asked a [similar question](https://support.xilinx.com/s/question/0D52E00007CfhWaSAJ/vitis-20211-failed-to-parse-the-xpfm-file?language=en_US).


Last year, the student used the "xilinx_u50_gen3x16_xdma_201920_3" Development Target Platform for their implementation of this github [project](https://github.com/bol-edu/2022-spring-nthu/blob/main/Final/gemm_hls/readme_u50.txt). 
However, the Xilinx official website only provides downloads for versions 2022.1 and 2022.2 .[here](https://www.xilinx.com/support/download/index.html/content/xilinx/en/downloadNav/alveo/u50.html)(step3)
![developmentTargetPlarform](https://github.com/bol-edu/HLS-Discussions/assets/98219588/280c9248-a5a5-42ce-b00d-4b1719984e29)

Is there a way to obtain an older version of the Development Target Platform?

Thank you.
許睿哲

**Comment by boledulab** [comment]
Hi,
Your default vitis have set to 2021.1 and you can find 201920_3.xpfm here.
/opt/xilinx/platforms/xilinx_u50_gen3x16_xdma_201920_3/xilinx_u50_gen3x16_xdma_201920_3.xpfm

**Comment by ray3210ray3210** [comment]
After solving the above problem, we encountered the following problem(Y2K22).
and **I don't have permission to update and copy files** to /opt/Xilinx/Vitis_HLS/2021.1/common/scripts/ and /opt/Xilinx/Vivado/2021.1/common/scripts/ 
Here is the error 
![exportIP](https://github.com/bol-edu/HLS-Discussions/assets/98219588/53ab3470-618e-4b68-808f-454ef8423700)


We found this [solution](https://support.xilinx.com/s/article/76960?language=en_US).
The solution is to download the "y2k22_patch-1.2.zip" file.
However, we found that the same folder already exists at /opt/Xilinx/y2k22_patch.
I followed the instructions in /opt/Xilinx/y2k22_patch/README. Here is the output.
![y2k_run](https://github.com/bol-edu/HLS-Discussions/assets/98219588/031d4d35-306d-4d41-a3d7-395eb56919c3)

but a successful output will look like [this](https://support.xilinx.com/s/article/76960?language=en_US)
![y2k22GoldenOutput](https://github.com/bol-edu/HLS-Discussions/assets/98219588/98543d48-70c6-4ab4-8377-6c7cb701876c)

I checked the two paths: /opt/Xilinx/Vitis_HLS/2021.1/common/scripts/  and /opt/Xilinx/Vivado/2021.1/common/scripts/ 

and found that the file "automg_patch_20220104.tcl" was not copied to those folders.

I think it is because  I don't have permission to perform the copy operation. 

Thank you.
許睿哲

**Comment by boledulab** [comment]
It should be updated, pls test your issue again.

hls02@HLS02:/opt/Xilinx$ export LD_LIBRARY_PATH=$PWD/Vivado/2021.1/tps/lnx64/python-3.8.3/lib/
hls02@HLS02:/opt/Xilinx$ Vivado/2021.1/tps/lnx64/python-3.8.3/bin/python3 y2k22_patch/patch.py
[2023-06-09] INFO: This script (version: 1.2) patches Xilinx Tools for HLS Y2k22 bug for the following release:
                2014.*, 2015.*, 2016.*, 2017.*, 2018.*, 2019.*, 2020.* and 2021.*
[2023-06-09] UPDATE: /opt/Xilinx/Vivado/2021.1/common/scripts
[2023-06-09] COPY: /opt/Xilinx/y2k22_patch/automg_patch_20220104.tcl  to /opt/Xilinx/Vivado/2021.1/common/scripts/automg_patch_20220104.tcl
[2023-06-09] UPDATE: /opt/Xilinx/Vitis_HLS/2021.1/common/scripts
[2023-06-09] COPY: /opt/Xilinx/y2k22_patch/automg_patch_20220104.tcl  to /opt/Xilinx/Vitis_HLS/2021.1/common/scripts/automg_patch_20220104.tcl

**Comment by ray3210ray3210** [comment]
Hi, for our final project, we are using an open-source library called "hlslib". During hardware emulation, we made some parameter adjustments, but it appears that the expected results are not being displayed. We found information on the library's [GitHub ](https://github.com/definelicht/hlslib/blob/a67e02f263d666e1c7ba649c6755025b5fdd38bf/README.md)page suggesting that it might be necessary to install the libc6-dev-i386 package on Ubuntu for hardware emulation. Can you help us with downloading this package? 

Thanks

![ubuntu package](https://github.com/bol-edu/HLS-Discussions/assets/98219588/c9812fc5-1ae3-4d9b-821d-05898d9f6cd7)

**Reply by boledulab** [reply]
The hls02 already installed libc6-dev-i386 (newest version 2.31-0ubuntu9.9).
